src/CharterManager.sol

```diff
@@ -222,6 +235,35 @@ contract CharterManagerImp {
         ManagedGemJoinLike(gemJoin).exit(urp, usr, amt);
     }
 
+    function move(address u, address dst, uint256 rad) external allowed(u) {
+        address urp = proxy[u];
+        require(urp != address(0), "CharterManager/non-existing-urp");
+
+        VatLike(vat).move(urp, dst, rad);
+    }
+
+    function roll(bytes32 srcIlk, bytes32 dstIlk, address src, address dst, uint256 srcDart) external allowed(src) allowed(dst) {
+        require(gate[srcIlk] == 1 && gate[dstIlk] == 1, "CharterManager/non-gated-ilks");
```

Comment:
Why is this necessary?

Reply:
Not strictly speaking, but I wanted to avoid users from trying to move from permissioned vaults (gate=1) to non permissioned ones (gate=0). You could argue that it would fail anyway since the vaults are not marked as rollable, but this makes it more clear.
This also allowed me to pass `1` as the last parameter to _validate() and avoid a "stack too deep" error.

Reply:
Yeah I'm not 100% convinced we should do it, but not opposed to it either. Let's leave it as it is.